lib/openstack_api/openstack_project.py

```diff
@@ -18,20 +18,27 @@ def create_project(
     :param project_details: A datastructure containing all the project details
     :return: A clouds project, or None if it was unsuccessful
     """
-    # domain_id can be blank, we will create a project in the default domain
+
     if not project_details.name:
         raise MissingMandatoryParamError("The project name is missing")
     if not project_details.email:
         raise MissingMandatoryParamError("The project contact email is missing")
     email_regex = r"^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$"
     if not re.match(email_regex, project_details.email):
         raise ValueError("The project contact email is invalid")
+    if not project_details.project_domain:
+        raise MissingMandatoryParamError("The domain for the project is missing")
 
     create_project_kwargs = {
         "name": project_details.name,
         "description": project_details.description,
         "is_enabled": project_details.is_enabled,
     }
+
+    # get domain id for the domain to use
+    domain_details = conn.identity.get_domain(project_details.project_domain)
```

Comment:
```suggestion
    domain_details = conn.identity.find_domain(project_details.project_domain)
```
This should be `find_domain` since `get_domain` requires us to already know the domain id to get the domain, `find_domain` can use either name or id

Reply:
I've addressed this in a follow-up commit